### PR TITLE
fix local tmp image path when remote url contains ?

### PR DESCRIPTION
**Pull request by pharosproduction**

### CODE REVIEW

File: lib/arc/file.ex [hunk 1]
```diff
@@ -15,7 +15,10 @@ defmodule Arc.File do
   # Given a remote file
   def new(remote_path = "http" <> _) do
     case save_file(remote_path) do
```
Comment (stavro) [inline lib/arc/file.ex]: I'd prefer to use the same parsing logic for determining the file name and extension (and only do it once).

Can you change this to:

```elixir
def new(remote_path = "http" <> _) do
  uri = URI.parse(remote_path)
  filename = Path.basename(uri.path)

  case save_file(uri, filename) do
    {:ok, local_path} -> %Arc.File{path: local_path, file_name: filename}
    :error -> {:error, :invalid_file_path}
  end
end

defp save_file(uri, filename) do
  local_path =
    generate_temporary_path()
    |> Kernel.<>(Path.extname(filename))

  # rest of this method...
end
```

File: lib/arc/file.ex [hunk 2]
```diff
@@ -70,12 +74,26 @@ defmodule Arc.File do
...
+      true -> {:error, :invalid_data}
     end
   end
 
   defp get_remote_path(remote_path) do
     case HTTPoison.get(remote_path) do
       {:ok, %{status_code: 200, body: body}} -> {:ok, body}
+      {:ok, %{status_code: 302, headers: headers}} ->
+        cond do
+          {"Location", url} = headers |> List.keyfind("Location", 0) -> new(url)
```
Comment (stavro) [inline lib/arc/file.ex]: HTTPoison has a `follow_redirect: true` option we can pass.

I'd much rather use that than build our own recursive function.  For example, we're not following `301` redirects.

File: lib/arc/file.ex [hunk 3]
```diff
@@ -70,12 +74,26 @@ defmodule Arc.File do
     case remote_file do
       {:ok, body} -> File.write(local_path, body)
       {:error, error} -> {:error, error}
+      other -> validate_arc_file(other)
+    end
+  end
+
+  defp validate_arc_file(struct) do
+    cond do
+      struct.__struct__ == Arc.File -> struct
```
Comment (stavro) [inline lib/arc/file.ex]: After we remove our own redirect logic, this `__struct__` and `other` matching should all go away

**Comment by pharosproduction** [comment]
great suggestion, it works.

**Pull request merged**